api/types.go

```diff
@@ -0,0 +1,47 @@
+package api
+
+type Context struct {
+	State DebuggerState `json:"state"`
```

Comment:
To take a simple example. Let's say I'm a client, and I'm displaying the current debugger state. I `GET /context` and receive:

```json
{
  "status": {
    "state": "Paused",
    "breakpoint": {  },
    "thread": {  }
  }
}
```
The UI is updated to reflect the current status of the debugger. The user decides to continue execution, so they `POST /context`:

```json
{
  "requested": {
    "state": "Continue"
  }
}
```

And let's say it's a slow program that hasn't yet reached the next breakpoint. The response will be:

```json
{
  "status": {
    "state": "Continuing",
    "thread": {  }
  }
}
```

Notice the lack of a current `breakpoint`. Let's say the program hits another breakpoint. The next `GET /context` returns:

```json
{
  "status": {
    "state": "Paused",
    "breakpoint": {  },
    "thread": {  }
  }
}
```

The pattern allows you to separate intent/actuality within a single state exchange to fit REST semantics cleanly without having to do things like POST command objects to an RPC-like endpoint. The same API also maps neatly to the event requirements (via the GET for new status).